src/app/app.cpp

```diff
@@ -699,6 +699,26 @@ bool CApplication::Create()
     return true;
 }
 
+void CApplication::Reload()
+{
+    m_sound->Create();
+    m_engine->ReloadAllTextures();
+    CThread musicLoadThread([this]()
```

Comment:
You just copy-pasted like 15 lines of code from CApplication::Create(), how about replacing it with a call to this function? Also, I'd rename it to `ReloadResources()` to avoid confusion, this is not reloading the whole application